addressbook/addressbook.sh

```
@@ -0,0 +1,267 @@
+#!/bin/ash
+
+# Functions
+format_echo()
+{
+	echo "$1" | sed 's/:/ | /g'
+}
+
+usage()
+{
+	echo "usage: addressbook [OPTIONS]"
+	echo
+	echo "OPTIONS"
+	echo "	-h"
+	echo "		Display usage"
+	echo "	-a"
+	echo "		Display all entries."
+	echo "	-s NAME PHONE EMAIL"
+	echo "		Save entrie with name, phone naumber and email address."
+	echo "	-f SEARCH"
+	echo "		Find entrie/entries by the given input."
+	echo "	-e INDEX FIELD_INDEX CHANGE"
+	echo "		Replace entrie's field with CHANGE"
+	echo "	-r INDEX"
+	echo "		Removes entrie by the given INDEX"
+}
+
+save_entrie()
+{
+	if  grep -q "${1}:${2}:${3}" data.txt
+	then
+		echo "Entrie already exists..."
+		echo
+	else
+		echo "$1:$2:$3" >> data.txt
+                echo Entrie has been saved...
+                echo
+
+	fi
+}
+
+get_all_entries()
+{
+	echo "Index	Name	Phone	  Email"
+	format_echo "`grep -n "$1" data.txt`"
+	echo
+}
+
+search_entrie()
+{
+	format_echo "`grep -n "$1" data.txt`"
+}
+
+remove_entrie()
+{
+	while :
+	do
+		if [ "$2" != "y" ]
+		then
+			echo "Are you sure?(y/n)"
+			read CHOICE
+		fi
+		if [ "$CHOICE" = "y" ]
+		then
+			INDEX=1
+        		while IFS= read LINE
+        		do
+				if [ "$INDEX" = "$1" ]
+				then
+                             		grep -v "$LINE" data.txt > temporary_file
+                               		mv temporary_file data.txt
+                               		if [ "$?" -eq "0" ]
+                               		then
+                                       		if [ "$2" != "y" ]
+						then
+							echo "Entrie has been removed..."
+							echo
+						fi
+                               		else
+                                       		echo "Error(couldn't remove entrie)"
+						echo
+                               		fi
+					break
+				fi
+				INDEX=`expr $INDEX + 1`
+	        	done < data.txt
+			break
+		elif [ "$CHOICE" = "n" ]
```

Anything that's not a precise `y` or `n` will silently skip. It's recommended to

* accept multiple input variants (eg.: y, Y, yes, Yes, YES, etc.)
* use an `else` clause to handle anything not considered a yes to be a no
* use an `else` clause to handle anything not considered a yes or no to be an input error